### PR TITLE
Add parameter for changing between testnet and devnet in e2e tests

### LINKED ISSUE #139: Running tests on starknet testnet
**Is your feature request related to a problem? Please describe.**

The problem comes from PR #131 where we have functionality that should be tested on testnet, but as correctly pointed out by @Arcticae these tests probably shouldn't be run on every commit as they introduce significant overhead.

**Describe the solution you'd like**

Some change in how these tests are run should be introduced.

**Describe alternatives you've considered**

Here are some ideas:
* Introduce manually triggered task that runs tests on testnet (most flexible, but requires most maintnance)
* Run testnet tests only on commits to development and master (automatic but may result in detecting problems after merge)
* Run testnet tests just before merging PRs (most optimal, but I don't know if it's event possible)

### PR DESCRIPTION
**Pull request by cptartur**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Added a `--net` parameter to e2e tests that allows changing between running tests on "devnet" and "testnet".

Will require further verification as running tests on testnet is super slow.

Should close #139 

* **What is the current behavior?** (You can also link to an open issue here)

Test can only be run on devnet

## **What is the new behavior (if this is a feature change)?**

Test can be run on devnet and testnet

## **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

### CODE REVIEW

**Comment by cptartur** [comment]
We should probably also add a option (perhaps a pytest mark) for some test to be run **only** on devnet/testnet

File: starknet_py/tests/e2e/conftest.py [hunk 1]
```diff
@@ -29,5 +37,35 @@ def run_devnet():
...
+def pytest_collection_modifyitems(config, items):
+    if config.getoption("--net") == "testnet":
+        skip_devnet = pytest.mark.skip()
+        for item in items:
+            if "run_on_testnet" not in item.keywords:
+                item.add_marker(skip_devnet)
+    else:
+        skip_testnet = pytest.mark.skip()
+        for item in items:
+            if "run_on_testnet" in item.keywords:
```
Comment (Solpatium) [inline starknet_py/tests/e2e/conftest.py:70]: Should this if be the same as above? Also, repeated code.

File: starknet_py/tests/e2e/conftest.py [hunk 2]
```diff
@@ -29,5 +37,35 @@ def run_devnet():
...
+                item.add_marker(skip_testnet)
```
Comment (Solpatium) [inline starknet_py/tests/e2e/conftest.py:71]: Simplify this

File: starknet_py/tests/e2e/conftest.py [hunk 3]
```diff
@@ -29,5 +37,28 @@ def run_devnet():
...
+@pytest.fixture(scope="module", autouse=True)
+def run_devnet():
+    devnet_port, proc = start_devnet()
     yield f"http://localhost:{devnet_port}"
     proc.kill()
+
+
+@pytest.fixture
+def run_net(request):
+    return request.config.getoption("--net")
```
Comment (Solpatium) [inline starknet_py/tests/e2e/conftest.py:52]: This might return "all". What it the purpose of `run_net`?

**Review by Solpatium — APPROVED** [review]

**Pull request merged**